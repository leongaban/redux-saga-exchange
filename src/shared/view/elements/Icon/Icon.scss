.icon {
  display: inline-block;

  svg {
    display: block;
    width: 100%;
    height: 100%;
  }
}
