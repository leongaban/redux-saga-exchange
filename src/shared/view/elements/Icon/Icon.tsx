import * as React from 'react';
import block from 'bem-cn';

import InlineSvg from 'svg-inline-react';

import './Icon.scss';

interface IProps {
  className: string;
  src: string;
}

const b = block('icon');

function Icon({ className, src }: IProps) {
  return <InlineSvg src={src} className={b.mix(className)()} />;
}

export default Icon;
