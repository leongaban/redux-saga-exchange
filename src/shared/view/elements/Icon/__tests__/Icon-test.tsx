import * as React from 'react';
import { shallow } from 'enzyme';
import Icon from '../Icon';

describe('(Shared) elements', () => {

    describe('Icon', () => {

        it('should render', () => {
            const component = shallow(<Icon className={'class1 class2'} src={'path/to/file'}/>);

            expect(component.debug()).toMatchSnapshot();
        });

    });

});
