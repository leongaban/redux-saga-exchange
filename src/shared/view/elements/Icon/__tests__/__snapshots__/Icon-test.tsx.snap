// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Shared) elements Icon should render 1`] = `"<InlineSVG src=\\"path/to/file\\" className=\\"icon class1 class2\\" element=\\"i\\" raw={false} />"`;
